src/Components/Search/state.ts

```diff
@@ -0,0 +1,48 @@
+import { Container } from "unstated"
+
+interface StateContainer {
+  hasEnteredPreviews: boolean
+  selectedPreviewIndex: number
+  previewHrefs: string[]
+}
+
+export class SearchBarState extends Container<StateContainer> {
+  state = {
+    hasEnteredPreviews: false,
+    selectedPreviewIndex: 0,
+    previewHrefs: [],
+  }
+
+  incrementSelectedPreviewIndex() {
+    this.setState({
+      selectedPreviewIndex: this.state.selectedPreviewIndex + 1,
+    })
+  }
+
+  reset() {
+    this.setState({
+      hasEnteredPreviews: false,
+      selectedPreviewIndex: 0,
+      previewHrefs: [],
+    })
+  }
+
+  enterPreview() {
+    this.setState({ hasEnteredPreviews: true, selectedPreviewIndex: 0 })
+  }
+
+  decrementOrLeavePreview() {
+    let selectedPreviewIndex = this.state.selectedPreviewIndex - 1
+    const hasEnteredPreviews = selectedPreviewIndex >= 0
+    selectedPreviewIndex = Math.max(0, selectedPreviewIndex)
+    this.setState({ hasEnteredPreviews, selectedPreviewIndex })
+  }
+
+  addHrefs(hrefs) {
```

Comment:
Rather than thinking in terms of `hrefs` maybe think in terms of items? 

```
registerPreviewItem(item)
```
and then later, navigate via `item.href`.